core/packages/contracts/contracts/XcmApp.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+import "./ChannelRegistry.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+// TODO: put encoding functions here
+//import "./XCMAppPallet.sol";
+
+/// @title Proxy
+/// @notice A simple pass through proxy.
+contract Proxy is Ownable {
+
+    /// @dev The signature of a function which executes xcm.
+    bytes4 private constant EXEC_XCM_FUNC = bytes4(keccak256("execute(bytes[])"));
+
+    /// @dev Calls into the XCM executor
+    /// @param _executor The address of the XCM executor.
+    /// @param _payload The XCM payload.
+    /// @return bool than indicates success of the call.
+    function execute(address _executor, bytes calldata _payload) external onlyOwner returns (bool) {
+        bytes memory encodedCall = bytes.concat(EXEC_XCM_FUNC, _payload);
+        (bool success, ) = _executor.delegatecall(encodedCall);
+        return success;
+    }
+}
+
+/// @title Xcm App
+/// @notice Implements XCM for the EVM.
+contract XcmApp {
+    /// @dev Channels which are allowed to call this app.
+    ChannelRegistry public immutable registry;
+
+    /// @dev A mapping or 32 byte hashed origins to proxy accounts.
+    mapping(bytes32 => Proxy) public proxies;
+
+    /// @dev A list of approved xcm executors.
+    mapping(bytes8 => address) public executors;
```

Comment:
Lets drop support for multiple executors (and adding and removing them), so we can focus on the most important bits first. Can re-introduce this later on